src/tabular-api-surface.ts

```diff
@@ -507,6 +524,24 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
         return false;
       };
 
+      if (this.bigtable.collectMetrics) {
```

Comment:
Could we have it do the callbacks every time, regardless of this variable?

We should assume almost all users have CSM enabled, since that's the default. And we'd be introducing a lot of extra branching if we check the metrics collection state every time

Reply:
Sure. I added the check out of concern for latency, but it is likely pretty minor.